src/dotty/tools/dotc/transform/TreeTransform.scala

```diff
@@ -174,32 +174,22 @@ object TreeTransforms {
  }
 
   /** A helper trait to transform annotations on MemberDefs */
-  trait AnnotationTransformer extends MiniPhaseTransform with InfoTransformer {
+  trait AnnotationTransformer extends MiniPhaseTransform with DenotTransformer {
```

Comment:
Currently that's not true, but I see you argument. I..e if we left out the self type transform from FirstTransform it would not be an InfoTransformer. And that transform is in FirstTransform only by accident, it could well be elsewhere.

*However* one could argue that every annotation transform should affect annotated types, and annotated types can be infos of symbols so in that sense every annotation transform *is* an info transform.

*But* that would require a complete typemap of every type in every info transform, and we know this leads to cyclic references. So we can't do this at present.

Bottom line: When in doubt decouple the functionalities.